neuralforecast/models/timesnet.py

```diff
@@ -0,0 +1,288 @@
+        return res
+
+# %% ../../nbs/models.timesnet.ipynb 8
+def FFT_for_Period(x, k=2):
+    # [B, T, C]
+    xf = torch.fft.rfft(x, dim=1)
+    # find period by amplitudes
+    frequency_list = abs(xf).mean(0).mean(-1)
+    frequency_list[0] = 0
+    _, top_list = torch.topk(frequency_list, k)
+    top_list = top_list.detach().cpu().numpy()
+    period = x.shape[1] // top_list
+    return period, abs(xf).mean(-1)[:, top_list]
+
+
+class TimesBlock(nn.Module):
+    def __init__(self, input_size, h, k, hidden_size, conv_hidden_size, num_kernels):
+        super(TimesBlock, self).__init__()
+        self.input_size = input_size
+        self.h = h
+        self.k = k
+        # parameter-efficient design
+        self.conv = nn.Sequential(
+            Inception_Block_V1(hidden_size, conv_hidden_size, num_kernels=num_kernels),
+            nn.GELU(),
+            Inception_Block_V1(conv_hidden_size, hidden_size, num_kernels=num_kernels),
+        )
+
+    def forward(self, x):
+        B, T, N = x.size()
+        period_list, period_weight = FFT_for_Period(x, self.k)
+
+        res = []
+        for i in range(self.k):
+            period = period_list[i]
+            # padding
+            if (self.input_size + self.h) % period != 0:
+                length = (((self.input_size + self.h) // period) + 1) * period
+                padding = torch.zeros(
+                    [x.shape[0], (length - (self.input_size + self.h)), x.shape[2]]
+                ).to(x.device)
+                out = torch.cat([x, padding], dim=1)
+            else:
+                length = self.input_size + self.h
+                out = x
+            # reshape
+            out = (
+                out.reshape(B, length // period, period, N)
+                .permute(0, 3, 1, 2)
+                .contiguous()
+            )
+            # 2D conv: from 1d Variation to 2d Variation
+            out = self.conv(out)
+            # reshape back
+            out = out.permute(0, 2, 3, 1).reshape(B, -1, N)
+            res.append(out[:, : (self.input_size + self.h), :])
+        res = torch.stack(res, dim=-1)
+        # adaptive aggregation
+        period_weight = F.softmax(period_weight, dim=1)
+        period_weight = period_weight.unsqueeze(1).unsqueeze(1).repeat(1, T, N, 1)
+        res = torch.sum(res * period_weight, -1)
+        # residual connection
+        res = res + x
+        return res
+
+# %% ../../nbs/models.timesnet.ipynb 10
+class TimesNet(BaseWindows):
+    """TimesNet
+
+    The TimesNet univariate model tackles the challenge of modeling multiple intraperiod and interperiod temporal variations.
+
+    The architecture has the following distinctive features:
+    - An embedding layer that maps the input sequence into a latent space.
+    - Transformation of 1D time seires into 2D tensors, based on periods found by FFT.
+    - A convolutional Inception block that captures temporal variations at different scales and between periods.
+
+    *Parameters:*<br>
```

Comment:
The HTML break tags are currently breaking our mintlify documentation, can you please make these with the [numpy style](https://numpydoc.readthedocs.io/en/latest/format.html). Also this notebook wasn't added, I guess it's nbs/models.timenet.ipynb

Reply:
thanks!